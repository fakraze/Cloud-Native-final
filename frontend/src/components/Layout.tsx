import React from 'react';
import { Outlet } from 'react-router-dom';
import { Header } from './Header';
import { Sidebar } from './Sidebar';
import { useAuthStore } from '../store/authStore';
import { useCart } from '../hooks/useOrder';

const Layout: React.FC = () => {
  const { user } = useAuthStore();
  
  // Initialize cart when layout loads
  useCart(user?.id || '1');

  return (
    <div className="flex h-screen bg-gray-50">
      <Sidebar />
      <div className="flex-1 flex flex-col overflow-hidden">
        <Header />
        <main className="flex-1 overflow-x-hidden overflow-y-auto bg-gray-50 p-4">
          <Outlet />
        </main>
      </div>
  </div>
  );
};

export { Layout };
