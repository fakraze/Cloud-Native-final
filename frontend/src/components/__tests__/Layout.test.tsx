import { describe, it, expect, beforeEach, vi } from 'vitest';
import { render } from '../../test/test-utils';
import { Layout } from '../Layout';

// Mock all the dependencies
vi.mock('../Header', () => ({
  Header: () => <div data-testid="header">Header Component</div>
}));

vi.mock('../Sidebar', () => ({
  Sidebar: () => <div data-testid="sidebar">Sidebar Component</div>
}));

vi.mock('../../store/authStore', () => ({
  useAuthStore: () => ({
    user: {
      id: '1',
      username: 'testuser',
      role: 'customer'
    }
  })
}));

vi.mock('../../hooks/useOrder', () => ({
  useCart: vi.fn()
}));

describe('Layout Component', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('should render layout with header and sidebar', () => {
    const { container } = render(<Layout />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have proper CSS structure', () => {
    const { container } = render(<Layout />);
    const layoutRoot = container.firstChild as HTMLElement;
    expect(layoutRoot.className).toContain('flex h-screen bg-gray-50');
    expect(layoutRoot).toMatchSnapshot();
  });
});
