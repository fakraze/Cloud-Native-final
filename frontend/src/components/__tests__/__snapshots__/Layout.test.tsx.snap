// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout Component > should have proper CSS structure 1`] = `
<div
  class="flex h-screen bg-gray-50"
>
  <div
    data-testid="sidebar"
  >
    Sidebar Component
  </div>
  <div
    class="flex-1 flex flex-col overflow-hidden"
  >
    <div
      data-testid="header"
    >
      Header Component
    </div>
    <main
      class="flex-1 overflow-x-hidden overflow-y-auto bg-gray-50 p-4"
    />
  </div>
</div>
`;

exports[`Layout Component > should render layout with header and sidebar 1`] = `
<div
  class="flex h-screen bg-gray-50"
>
  <div
    data-testid="sidebar"
  >
    Sidebar Component
  </div>
  <div
    class="flex-1 flex flex-col overflow-hidden"
  >
    <div
      data-testid="header"
    >
      Header Component
    </div>
    <main
      class="flex-1 overflow-x-hidden overflow-y-auto bg-gray-50 p-4"
    />
  </div>
</div>
`;
